doc/source/CodeFormatting.rst

```diff
@@ -0,0 +1,106 @@
+Style Formatting
+================
+
+The Grackle repository is configured with tools to enforce checks (such as running the linter or applying code-formatting) on the various files in the repository.
+
+The primary checks pertain to python code: all python code is linted and all new python files are formatted by automated tools (for the time being, code in older files will not be formatted to avoid merge-conflicts).
+
+.. note::
+
+   In the near future (see :gh-pr:`222`), we will start formatting C/C++ code.
+
+Because we realize that these tools may seem overwhelming or complicated, we have configured the repository to try to simplify the experience to the greatest extent possible.
+All of these checks are managed by the `pre-commit <https://pre-commit.com/>`__ software.
+We will discuss how to automatically invoke these checks as part of a PR (no local installation is required) or on your local machine down below.
+
+Automatic Checks (No installation required)
+-------------------------------------------
+
+At a basic level, you don't have to worry about manually invoking any of these tools on your machine.
+In fact, you are free to entirely ignore the existence of these tools until it comes time to submit a Pull Request.
+When you submit a Pull Request (and whenever you update it), various forms of continuous integration are triggered.
+
+For the present discussion, the `pre-commit.ci <https://pre-commit.ci/>`__ continuous integration tool is of primary relevance.\ [#f1]_
+This tool executes all of the formatting tools and if your submission doesn't satisfy all of the requirements, it will fail and report each problems.
```

Comment:
```suggestion
This tool executes all of the formatting tools and if your submission doesn't satisfy all of the requirements, it will fail and report each problem.
```